JSONObject.java

```diff
@@ -1828,14 +1822,23 @@ public static Object wrap(Object object) {
      * @return The writer.
      * @throws JSONException
      */
-    public Writer write(Writer writer) throws JSONException {
+    public <T extends Appendable> T write(T writer) throws JSONException {
         return this.write(writer, 0, 0);
     }
 
-    static final Writer writeValue(Writer writer, Object value,
+    static <T extends Appendable> T writeValue(T writer, Object value)
```

Comment:
New method used by JSONObject.valueToString(), JSONArray.join(), and JSONWriter.appendValue(). Simple delegation.